src/extension.ts

```diff
@@ -75,6 +75,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         vscode.workspace.createFileSystemWatcher('**/*.tfvars'),
       ],
     },
+    diagnosticCollectionName: 'HashiCorpTerraform',
```

Comment:
[It looks like](https://github.com/search?q=diagnosticCollectionName&type=code) there isn't any conventions for the name. Some extensions choose to reflect the fact that diagnostics come from the LS and some don't, some use spaces, some camelCase. So ... **TL;DR LGTM**

On a separate note, I would have hoped that VS Code separates out diagnostics produced by different extensions "out of the box" but I suppose I have to adjust my expectations. 😄